apps/dashboard/src/components/workflow-editor/test-workflow/test-workflow-logs-sidebar.tsx

```diff
@@ -55,33 +57,48 @@ export const TestWorkflowLogsSidebar = ({ transactionId, workflow }: TestWorkflo
     }
   }, [activityId]);
 
-  // Reset refetch when transaction ID changes
+  const handleTransactionIdChange = useCallback((newTransactionId: string) => {
+    setIsTransactionChanging(true);
+    setTransactionId(newTransactionId);
+    setParentActivityId(undefined);
+  }, []);
+
+  useEffect(() => {
+    if (activity) {
+      setIsTransactionChanging(false);
+    }
+  }, [activity]);
+
   useEffect(() => {
-    if (!transactionId) {
+    if (!props.transactionId) {
       return;
     }
 
     setShouldRefetch(true);
-    setParentActivityId(undefined);
-  }, [transactionId]);
+    setTransactionId(props.transactionId);
+  }, [props.transactionId]);
 
   return (
     <aside className="flex h-full max-h-full flex-1 flex-col overflow-auto">
       {transactionId ? (
         <>
           <ActivityPanel>
-            {isPending ? (
+            {isPending || isTransactionChanging ? (
               <ActivitySkeleton />
             ) : error || !activity ? (
               <ActivityError />
             ) : (
               <React.Fragment key={activityId}>
                 <ActivityHeader title={activity.template?.name} className="h-[49px] border-t-0" />
                 <ActivityOverview activity={activity} />
-                <ActivityLogs activity={activity} onActivitySelect={setParentActivityId} />
+                <ActivityLogs
+                  activity={activity}
+                  onActivitySelect={setParentActivityId}
+                  onTransactionIdChange={handleTransactionIdChange}
```

Comment:
I'm missing the similar handling logic in the `subscriber-activity-drawer.tsx`

Reply:
you are amazing! i totally forgot that we have an activity feed in the subscriber page as well!